Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/site-packages/nbclient/client.py", line 1314, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
  File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/site-packages/nbclient/client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# Calcular o índice de correlação Spearman
correlacao_spearman = df_sem_outliers['subsidio_valor_mensal'].corr(
    df_sem_outliers['qtd_fam_aptas_ou_acolhendo'],
    method='spearman'
)
print(f"Índice de correlação (Spearman): {correlacao_spearman:.2f}")
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mModuleNotFoundError[0m                       Traceback (most recent call last)
Cell [0;32mIn[17], line 2[0m
[1;32m      1[0m [38;5;66;03m# Calcular o índice de correlação Spearman[39;00m
[0;32m----> 2[0m correlacao_spearman [38;5;241m=[39m [43mdf_sem_outliers[49m[43m[[49m[38;5;124;43m'[39;49m[38;5;124;43msubsidio_valor_mensal[39;49m[38;5;124;43m'[39;49m[43m][49m[38;5;241;43m.[39;49m[43mcorr[49m[43m([49m
[1;32m      3[0m [43m    [49m[43mdf_sem_outliers[49m[43m[[49m[38;5;124;43m'[39;49m[38;5;124;43mqtd_fam_aptas_ou_acolhendo[39;49m[38;5;124;43m'[39;49m[43m][49m[43m,[49m
[1;32m      4[0m [43m    [49m[43mmethod[49m[38;5;241;43m=[39;49m[38;5;124;43m'[39;49m[38;5;124;43mspearman[39;49m[38;5;124;43m'[39;49m
[1;32m      5[0m [43m)[49m
[1;32m      6[0m [38;5;28mprint[39m([38;5;124mf[39m[38;5;124m"[39m[38;5;124mÍndice de correlação (Spearman): [39m[38;5;132;01m{[39;00mcorrelacao_spearman[38;5;132;01m:[39;00m[38;5;124m.2f[39m[38;5;132;01m}[39;00m[38;5;124m"[39m)

File [0;32m/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/site-packages/pandas/core/series.py:2977[0m, in [0;36mSeries.corr[0;34m(self, other, method, min_periods)[0m
[1;32m   2974[0m other_values [38;5;241m=[39m other[38;5;241m.[39mto_numpy(dtype[38;5;241m=[39m[38;5;28mfloat[39m, na_value[38;5;241m=[39mnp[38;5;241m.[39mnan, copy[38;5;241m=[39m[38;5;28;01mFalse[39;00m)
[1;32m   2976[0m [38;5;28;01mif[39;00m method [38;5;129;01min[39;00m [[38;5;124m"[39m[38;5;124mpearson[39m[38;5;124m"[39m, [38;5;124m"[39m[38;5;124mspearman[39m[38;5;124m"[39m, [38;5;124m"[39m[38;5;124mkendall[39m[38;5;124m"[39m] [38;5;129;01mor[39;00m [38;5;28mcallable[39m(method):
[0;32m-> 2977[0m     [38;5;28;01mreturn[39;00m [43mnanops[49m[38;5;241;43m.[39;49m[43mnancorr[49m[43m([49m
[1;32m   2978[0m [43m        [49m[43mthis_values[49m[43m,[49m[43m [49m[43mother_values[49m[43m,[49m[43m [49m[43mmethod[49m[38;5;241;43m=[39;49m[43mmethod[49m[43m,[49m[43m [49m[43mmin_periods[49m[38;5;241;43m=[39;49m[43mmin_periods[49m
[1;32m   2979[0m [43m    [49m[43m)[49m
[1;32m   2981[0m [38;5;28;01mraise[39;00m [38;5;167;01mValueError[39;00m(
[1;32m   2982[0m     [38;5;124m"[39m[38;5;124mmethod must be either [39m[38;5;124m'[39m[38;5;124mpearson[39m[38;5;124m'[39m[38;5;124m, [39m[38;5;124m"[39m
[1;32m   2983[0m     [38;5;124m"[39m[38;5;124m'[39m[38;5;124mspearman[39m[38;5;124m'[39m[38;5;124m, [39m[38;5;124m'[39m[38;5;124mkendall[39m[38;5;124m'[39m[38;5;124m, or a callable, [39m[38;5;124m"[39m
[1;32m   2984[0m     [38;5;124mf[39m[38;5;124m"[39m[38;5;124m'[39m[38;5;132;01m{[39;00mmethod[38;5;132;01m}[39;00m[38;5;124m'[39m[38;5;124m was supplied[39m[38;5;124m"[39m
[1;32m   2985[0m )

File [0;32m/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/site-packages/pandas/core/nanops.py:85[0m, in [0;36mdisallow.__call__.<locals>._f[0;34m(*args, **kwargs)[0m
[1;32m     81[0m     [38;5;28;01mraise[39;00m [38;5;167;01mTypeError[39;00m(
[1;32m     82[0m         [38;5;124mf[39m[38;5;124m"[39m[38;5;124mreduction operation [39m[38;5;124m'[39m[38;5;132;01m{[39;00mf_name[38;5;132;01m}[39;00m[38;5;124m'[39m[38;5;124m not allowed for this dtype[39m[38;5;124m"[39m
[1;32m     83[0m     )
[1;32m     84[0m [38;5;28;01mtry[39;00m:
[0;32m---> 85[0m     [38;5;28;01mreturn[39;00m [43mf[49m[43m([49m[38;5;241;43m*[39;49m[43margs[49m[43m,[49m[43m [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43mkwargs[49m[43m)[49m
[1;32m     86[0m [38;5;28;01mexcept[39;00m [38;5;167;01mValueError[39;00m [38;5;28;01mas[39;00m e:
[1;32m     87[0m     [38;5;66;03m# we want to transform an object array[39;00m
[1;32m     88[0m     [38;5;66;03m# ValueError message to the more typical TypeError[39;00m
[1;32m     89[0m     [38;5;66;03m# e.g. this is normally a disallowed function on[39;00m
[1;32m     90[0m     [38;5;66;03m# object arrays that contain strings[39;00m
[1;32m     91[0m     [38;5;28;01mif[39;00m is_object_dtype(args[[38;5;241m0[39m]):

File [0;32m/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/site-packages/pandas/core/nanops.py:1614[0m, in [0;36mnancorr[0;34m(a, b, method, min_periods)[0m
[1;32m   1611[0m a [38;5;241m=[39m _ensure_numeric(a)
[1;32m   1612[0m b [38;5;241m=[39m _ensure_numeric(b)
[0;32m-> 1614[0m f [38;5;241m=[39m [43mget_corr_func[49m[43m([49m[43mmethod[49m[43m)[49m
[1;32m   1615[0m [38;5;28;01mreturn[39;00m f(a, b)

File [0;32m/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/site-packages/pandas/core/nanops.py:1629[0m, in [0;36mget_corr_func[0;34m(method)[0m
[1;32m   1627[0m     [38;5;28;01mreturn[39;00m func
[1;32m   1628[0m [38;5;28;01melif[39;00m method [38;5;241m==[39m [38;5;124m"[39m[38;5;124mspearman[39m[38;5;124m"[39m:
[0;32m-> 1629[0m     [38;5;28;01mfrom[39;00m [38;5;21;01mscipy[39;00m[38;5;21;01m.[39;00m[38;5;21;01mstats[39;00m [38;5;28;01mimport[39;00m spearmanr
[1;32m   1631[0m     [38;5;28;01mdef[39;00m [38;5;21mfunc[39m(a, b):
[1;32m   1632[0m         [38;5;28;01mreturn[39;00m spearmanr(a, b)[[38;5;241m0[39m]

[0;31mModuleNotFoundError[0m: No module named 'scipy'

